src/main/java/EOorg/EOeolang/EOthreads/EOmutex$EOacquire$EOrelease$EOφ.java

```diff
@@ -59,12 +59,16 @@
             new AtComposite(
                 this,
                 rho -> {
-                    final Phi mutex = rho.attr("ρ").get().attr("ρ").get().attr("ρ").get();
+                    final Phi acquire = rho.attr("ρ").get().attr("ρ").get();
+                    final Phi mutex = acquire.attr("ρ").get();
                     final long releases = new Dataized(
                         rho.attr("ρ").get().attr("releases").get()
                     ).take(Long.class);
-                    final Semaphore semaphore = Semaphores.INSTANCE.get(mutex);
-                    semaphore.release((int) releases);
+                    synchronized (acquire) {
```

Comment:
@levBagryansky I have a feeling that this synchronization won't help here, since every time you `get()` an object from `rho`, it's a new object. Did you try to test your code for thread-safety? Check this: https://www.yegor256.com/2018/03/27/how-to-test-thread-safety.html